### PR TITLE
feat: Add features management tooling

### PR DESCRIPTION
**Pull request by KirilMihaylov**
Introduces the `cargo-features` tool, whose purpose is to be able to list the features and make intersections between the passed and existing features of a package.
Due to how the `cargo` works, there are two ways to invoke a tool. Directly, `cargo-features`, or indirectly through, `cargo features`. Both variants are supported in order to provide better ergonomics and ease of use.

Additional changes:
* The build system is switched to use this new mechanism to filter-out features that don't exist, e.g. `contract` feature on `rewards_dispatcher` and `leaser`.
* Marked `/platform` directory as mount point for a volume.
* Removed temporarily added, unused, feature `contract` from `rewards_dispatcher` and `leaser`

### CODE REVIEW

**Comment by KirilMihaylov** [comment]
Optimized build: https://github.com/nolus-protocol/nolus-money-market/actions/runs/7112849507

File: Containerfile [hunk 1]
```diff
@@ -10,8 +10,24 @@ VOLUME ["/artifacts"]
 
 VOLUME ["/code"]
```
[the remark this answers is not in the document]
Comment (KirilMihaylov) [reply Containerfile:11]: Replied in DMs.
Tl;dr: `/code` is how the workspace /`platform` & `protocol`/ is selected.

**Review by gmanev7 — APPROVED** [review]

**Comment by KirilMihaylov** [comment]
Optimized build workflow: https://github.com/nolus-protocol/nolus-money-market/actions/runs/7126087248
